### ISSUE TITLE
Fix Reduce Damage icon

### ISSUE THREAD

**Issue by by003** [vcmi/vcmi#3471]
**Describe the bug**
[#2183](https://github.com/vcmi/vcmi/issues/2183)

I cannot modify any cpp files in my client. That's the solution.

Original codes in **vcmi/lib/CBonusTypeHandler.cpp**

```
case BonusType::GENERAL_DAMAGE_REDUCTION:
	{
		if (bonus->subtype == BonusCustomSubtype::damageTypeMelee)
			fileName = "DamageReductionMelee.bmp";

		if (bonus->subtype == BonusCustomSubtype::damageTypeRanged)
			fileName = "DamageReductionRanged.bmp";

		break;
	}
```

Add a condition:

```
case BonusType::GENERAL_DAMAGE_REDUCTION:
	{
		if (bonus->subtype == BonusCustomSubtype::damageTypeMelee)
			fileName = "DamageReductionMelee.bmp";

		if (bonus->subtype == BonusCustomSubtype::damageTypeRanged)
			fileName = "DamageReductionRanged.bmp";
                
                if (bonus->subtype == BonusCustomSubtype::damageTypeAll)
			fileName = "DamageReductionAll.bmp";

		break;
	}
```

Then download this icon, and save as DamageReductionAll.bmp in vcmi-extra.
![DamageReductionAll](https://github.com/vcmi/vcmi/assets/127420617/9257fc04-ab66-406f-ae80-848f5242a5c2)
[DamageReductionAll.zip](https://github.com/vcmi/vcmi/files/13881561/DamageReductionAll.zip)

Check it.
![image](https://github.com/vcmi/vcmi/assets/127420617/ff68521e-8398-44d6-b611-3837ca2ec51b)

**Version**
 - OS: Windows
 - Version: 1.42

**Comment by by003** [comment]
@IvanSavenko Please check. I almost forget it while last day I saw someone mentioned it in Discord.

**Comment by by003** [comment]
Maybe consider to fix this in 1.4.3 @IvanSavenko

**Comment by IvanSavenko** [comment]
Additional check added in 1.4.3
